File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
const minCostClimbingStairs = (cost) => {
  for (let index = cost.length - 3; index >= 0; index--) {
    cost[index] += Math.min(cost[index + 1], cost[index + 2]);
  }
  return Math.min(cost[0], cost[1]);
};

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.cpp
class Solution
{
public:
    int minCostClimbingStairs(vector<int> &cost)
    {
        int one = 0;
        int two = 0;
        for (int i = 2; i <= cost.size(); i++)
        {
            int temp = one;
            one = min(one + cost[i - 1], two + cost[i - 2]);
            two = temp;
        }
        return one;
    }
};